src/uct/api/uct.h

```diff
@@ -1603,6 +1608,35 @@ UCT_INLINE_API ucs_status_t uct_ep_am_zcopy(uct_ep_h ep, uint8_t id, void *heade
     return ep->iface->ops.ep_am_zcopy(ep, id, header, header_length, iov, iovcnt, comp);
 }
 
+
+/**
+ * @ingroup UCT_AM
+ * @brief Register non-contiguous memory.
```

Comment:
Please add more description about this function, including:
 - how different types of memory should be described
 - completion semantics